dali/operators/image/paste/multipaste.cc

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/paste/multipaste.h"
+#include "dali/kernels/imgproc/paste/paste.h"
+#include "dali/core/tensor_view.h"
+
+namespace dali {
+namespace {
+
+template <typename Out, typename In>
+using TheKernel = kernels::PasteCpu<Out, In>;
+
+}  // namespace
+
+
+DALI_SCHEMA(MultiPaste)
+.DocStr(R"code(Performs multiple (K * batch_size) pastes from image batch to an output
+
+This operator can also change the type of data.)code")
+
+.NumInput(5, 6)
+.InputDox(0, "images", "3D TensorList", R"code(Batch of input images.
+
+Assumes HWC layout.)code")
+.InputDox(1, "in_ids", "1D TesorList of shape [K] and type int",
+R"code(Indexes from what inputs to paste data in each iteration.)code")
+.InputDox(2, "in_anchors", "2D TesorList of shape [K, 2] and type int",
+R"code(Absolute values of LU corner of the selection for each iteration.)code")
+.InputDox(3, "in_shapes", "2D TesorList of shape [K, 2] and type int",
+R"code(Absolute values of size of the selection for each iteration.)code")
+.InputDox(4, "out_anchors", "2D TesorList of shape [K, 2] and type int",
+R"code(Absolute values of LU corner of the paste for each iteration.)code")
+.InputDox(5, "out_ids", "1D TesorList of shape [K] and type int",
+R"code(Indexes to what outputs to paste data in each iteration.
+If ommitted, i-th tensor pastes to i-th output. )code")
```

Comment:
All of these should be tensor arguments, not regular inputs.

Reply:
Changed in the new commit, tests were modified accordingly and pass.